src/components/TopTabBar.js

```diff
@@ -0,0 +1,178 @@
+// @flow
+import React, { PureComponent } from "react";
+import { Animated, StyleSheet, View } from "react-native";
+import type { TabScene, _TabBarBottomProps } from "react-navigation";
+import Text from "./Text";
+import Touchable from "./Touchable";
+import {
+  filterButtonTextColor,
+  filterButtonsBgColor,
+  whiteColor,
+  topTabBarDividerColor
+} from "../constants/colors";
+
+class TopTabBar extends PureComponent<_TabBarBottomProps> {
+  constructor() {
+    super();
+    this.tabBarWidth = new Animated.Value(1);
+    this.tabLayouts = {};
+    this.activeTabX = new Animated.Value(0);
+    this.activeTabWidth = new Animated.Value(0);
+  }
+
+  componentDidUpdate() {
+    this.updateActiveTabLine();
+  }
+
+  handleTabPress = (key: string) => {
+    this.props.jumpTo(key);
+  };
+
+  tabBarLayout: any;
+  tabBarWidth: Animated.Value;
+  tabLayouts: { [number]: any };
+  activeTabX: Animated.Value;
+  activeTabWidth: Animated.Value;
+
+  updateActiveTabLine = () => {
```

Comment:
Maybe add a TODO comment just to indicate that this is duplicated in `NavigationTabBar`